Sources/TuistCache/Utilities/CacheFrameworkBuilder.swift

```diff
@@ -93,33 +97,63 @@ public final class CacheFrameworkBuilder: CacheArtifactBuilding {
         }
         let scheme = target.name.spm_shellEscaped()
 
-        // Create temporary directories
-        let builtProductsDirFingerprint = String.random()
         logger.notice("Building .framework for \(target.name)...", metadata: .section)
 
         let sdk = self.sdk(target: target)
         let configuration = "Debug" // TODO: Is it available?
 
         let arguments = try self.arguments(target: target,
                                            sdk: sdk,
-                                           configuration: configuration,
-                                           builtProductsDirFingerprint: builtProductsDirFingerprint)
+                                           configuration: configuration)
+
         try xcodebuild(
             projectTarget: projectTarget,
             scheme: scheme,
             target: target,
             arguments: arguments
         )
 
-        try exportFrameworksAndDSYMs(into: outputDirectory,
-                                     target: target,
-                                     builtProductsDirFingerprint: builtProductsDirFingerprint)
+        let buildDirectory = try self.buildDirectory(for: projectTarget,
+                                                     target: target,
+                                                     configuration: configuration,
+                                                     sdk: sdk)
+
+        try exportFrameworksAndDSYMs(from: buildDirectory,
+                                     into: outputDirectory,
+                                     target: target)
+    }
+
+    fileprivate func buildDirectory(for projectTarget: XcodeBuildTarget,
+                                    target: Target,
+                                    configuration: String,
+                                    sdk: String) throws -> AbsolutePath
+    {
+        let projectPath = projectTarget.path
+        let pathString = projectPath.pathString
+
+        if let existing = projectPathHashes[pathString] {
+            logger.info("Found previously calculated build directory for \(target.name) at: \(existing)")
```

Comment:
ups, I sent my review to the other branch, moving my comments here:
I'd suggest removing some of these logs as I see them more for debugging purposes

Reply:
Removed, thanks for catching it